### PR TITLE
Data encryption is not follow the AWS agreement

### PR DESCRIPTION
**Pull request by chuang-he**
https://github.com/ceph/ceph/pull/15994

### CODE REVIEW

**Comment by chuang-he** [comment]
@cbodley

**Comment by chuang-he** [comment]
@aclamk

**Comment by aclamk** [comment]
@chuang-he It would be nice if test `test_sse_kms_method_head()` also could test refusal to service if crypto headers are provided, like:
`res = _make_request('HEAD', bucket, key, authenticated=True, request_headers=sse_kms_client_headers)
eq(res.status, 400)`

**Comment by chuang-he** [comment]
@aclamk done

**Comment by yehudasa** [comment]
@chuang-he can you rebase?

**Comment by chuang-he** [comment]
@yehudasa  done

File: s3tests/functional/test_s3.py [hunk 1]
```diff
@@ -8768,3 +8746,40 @@ def test_bucket_policy_set_condition_operator_end_with_IfExists():
...
+    e = assert_raises(boto.exception.S3ResponseError, key.get_contents_as_string, headers=sse_kms_client_headers)
+    eq(e.status, 400)
+
+
+@attr(resource='object')
+@attr(method='head')
+@attr(operation='write encrypted with SSE-KMS and head with SSE-KMS')
+@attr(assertion='operation fails')
+@attr('encryption')
+def test_sse_kms_method_head():
```
Comment (cbodley) [inline s3tests/functional/test_s3.py]: there is already a `test_sse_kms_method_head()` above that verifies that HEAD requests return the right encryption headers. can you just add that last bit that checks for status=400 to the test above?

**Comment by chuang-he** [comment]
@cbodley  done

**Comment by cbodley** [comment]
@yehudasa could you please merge? we'll also need to get this on `ceph-master` so we can merge https://github.com/ceph/ceph/pull/15994

**Comment by cbodley** [comment]
@chuang-he can you please rebase?

**Comment by chuang-he** [comment]
@cbodley done

**Comment by cbodley** [comment]
@yehudasa can you please merge? https://github.com/ceph/ceph/pull/15994 is still blocked on getting this into ceph-master

**Pull request merged**